containers/ecr-viewer/src/app/api/services/database_repo.ts

```diff
@@ -0,0 +1,360 @@
+  }
+
+  if (criteria.data_source) {
+    query = query.where("data_source", "=", criteria.data_source);
+  }
+
+  if (criteria.fhir_reference_link) {
+    query = query.where(
+      "fhir_reference_link",
+      "=",
+      criteria.fhir_reference_link,
+    );
+  }
+
+  if (criteria.patient_name_last !== undefined) {
+    query = query.where(
+      "patient_name_last",
+      criteria.patient_name_last === null ? "is" : "=",
+      criteria.patient_name_last,
+    );
+  }
+
+  if (criteria.patient_birth_date) {
+    query = query.where("patient_birth_date", "=", criteria.patient_birth_date);
+  }
+
+  if (criteria.date_created) {
+    query = query.where("date_created", "=", criteria.date_created);
+  }
+
+  return await query.selectAll().execute();
+}
+
+/**
+ * Creates an eICR object
+ * @async
+ * @function createEcr
+ * @param ecr - the NewECR to be persisted
+ * @returns the created eICR object
+ */
+export async function createEcr(ecr: NewECR | null) {
+  if (!ecr || ecr === null) {
+    return console.error("eICR Data is required.");
+  }
+  try {
+    return await (db as Kysely<Core>)
+      .insertInto("ecr_data")
+      .values(ecr)
+      .returningAll()
+      .executeTakeFirstOrThrow();
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/**
+ * Updates an eICR object
+ * @async
+ * @function updateEcr
+ * @param eICR_ID - the eICR_ID of the eCR to be updated
+ * @param updateWith - the ECRUpdate to be applied to the existing record
+ * @returns the updated eICR object
+ */
+export async function updateEcr(eICR_ID: string | null, updateWith: ECRUpdate) {
+  await (db as Kysely<Core>)
+    .updateTable("ecr_data")
+    .set(updateWith)
+    .where("eICR_ID", "=", eICR_ID)
+    .execute();
+}
+
+/**
+ * Deletes an eICR object
+ * @async
+ * @function deleteEcr
+ * @param eICR_ID - the eICR_ID of the record to be deleted
+ * @returns the deleted eICR object
+ */
+export async function deleteEcr(eICR_ID: string | null) {
+  const ecr = await findEcrById(eICR_ID);
+
+  if (ecr) {
+    await (db as Kysely<Core>)
+      .deleteFrom("ecr_data")
+      .where("eICR_ID", "=", eICR_ID)
+      .execute();
+  }
+
+  return ecr;
+}
+
+// ECR_RR_CONDITIONS
+
+/**
+ * Finds an eCR condition by its ID
+ * @async
+ * @function findEcrConditionById
+ * @param id - the UUID of the eCR condition being looked up
+ * @returns an eCR condition object
+ */
+export async function findEcrConditionById(id: string) {
+  return await (db as Kysely<Core>)
+    .selectFrom("ecr_rr_conditions")
+    .where("uuid", "=", id)
+    .selectAll()
+    .executeTakeFirst();
+}
+
+/**
+ * Finds an eCR condition by its criteria
+ * @async
+ * @function findEcrCondition
+ * @param criteria - the Partial<ECRConditions> filtering criteria to be applied
+ * @returns an eCR condition object
+ */
+export async function findEcrCondition(criteria: Partial<ECRConditions>) {
+  let query = (db as Kysely<Core>).selectFrom("ecr_rr_conditions");
+
+  if (criteria.uuid) {
```

Comment:
same comments/thoughts as `findEcr` (and probably any future `findBlah` functions)